main.go

```diff
@@ -32,6 +33,25 @@ import (
 	"github.com/zmap/zgrab/ztools/ztls"
 )
 
+type port struct {
+	value uint16
+	set   bool
```

Comment:
It seems heavy handed to keep a bool around for the lifetime of the port just to identify if it is set. You could accomplish this with a sentential value, a ptr, or separate data structure that only lives the lifetime of the configuration.